Update templates to use new data2text feature

manuelfuenmayor opened the pull request:
Pending update in `modules/module_annex_change_history.adoc` file

UPDATE:

I've finished the updates, but I still need a solution for accessing to `documents/iso-10303-*/changes.yaml` data via `loadfile`. More details in https://github.com/metanorma/metanorma-plugin-datastruct/issues/67. Ping @kwkwan.

manuelfuenmayor committed:
Update yaml2text blocks to use new data2text approach

ronaldtse commented on resources/resource_annex_change_history.adoc:
```diff
@@ -15,10 +15,10 @@ Unless otherwise specified all modifications are upwardly compatible to the prev
 ------
 {% liquid
-assign path_prefix = "../../schemas/resources/"
+assign path_prefix = "schemas/resources/"
```
@kwkwan can we fix data2text to respect the location of the file path?

manuelfuenmayor replied:
@ronaldtse yes, that would be really useful. For module documents, I need to have access to a `changes.yaml` file that is placed at the same location as `document.adoc`.

Maybe we could do something like this for `loadfile` macro?

```
// fetch respect the document location (e.g. document.adoc)
{% assign data = path | loadfile %}

// fetch respect the root folder where command is executed
{% assign data = path | loadfile: "." %}
```

manuelfuenmayor replied:
@ronaldtse I think `data2text` is indeed fixed to the file location (which in this case is `documents/iso-10303-*/` folder).
The reason I redefined the indicated string is because of `loadfile: "."` macro. Given that it is taking the root folder (where command is executed) as reference. Hence, the change.

However, for the `modules/module_annex_change_history.adoc` template, I need to have access to a `changes.yaml` file that is placed at the same location as `document.adoc`.

Maybe we could do something like this for `loadfile` macro?

```
// fetch respect the document location (e.g. document.adoc)
{% assign data = path | loadfile %}

// fetch respect the root folder where command is executed
{% assign data = path | loadfile: "." %}
```

manuelfuenmayor committed:
Use new data2text in Changes history, module documents

Pending solution of: https://github.com/metanorma/metanorma-plugin-datastruct/issues/67
regarding the access to "changes.yaml" file via `loadfile`, in document folder.

manuelfuenmayor committed:
Fetch changes.yaml data from data2text heading

Considerations:
- If changes.yaml does not exist at doument.adoc location,
  compilation error will be raised.
- If any of these files does not exist at schema location:
  arm.changes.yaml, mim.changes.yaml or mapping.changes.yaml,
  compilation warning will be raised.

manuelfuenmayor committed:
Module docs: Add comments on the design of Change History template

manuelfuenmayor commented:
@ronaldtse I think this PR is ready to go.

Some considerations to take into account
- If `changes.yaml` file does not exist at `document.adoc` location, compilation error is raised.
  (I will raise a PR in ISO 10303 repository to introduce empty `changes.yaml` files where needed.)
- If any of these files: `arm.changes.yaml`, `mim.changes.yaml`, and `mapping.changes.yaml`, don't exist at schema location, a warning message like this one below will be raised:
   ```
   File referenced in [{data|yaml|json}2text] block not found: C:/Users/Manuel/Documents/Metanorma/iso-10303-srl/schemas/ modules/constructive_solid_geometry_3d/mapping.changes.yaml
   [metanorma-plugin-lutaml] Liquid render error: Liquid error: internal
   ```

ronaldtse commented:
Thank you @manuelfuenmayor !
